opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/RuntimeEnvironment.java

```diff
@@ -542,10 +568,9 @@ public String getUrlPrefix() {
      */
     public String getCtags() {
         String value;
-        return ctags != null ? ctags : (value =
-            threadConfig.get().getCtags()) != null ? value :
-            System.getProperty(SYSTEM_CTAGS_PROPERTY,
-            "ctags");
+        return ctags != null ? ctags :
+                (value = (String)getConfigurationValue("ctags")) != null ? value :
+                System.getProperty(SYSTEM_CTAGS_PROPERTY,"ctags");
     }
 
     /**
```

Comment:
setting ctags doesn't need locking?

Reply:
I think that's fine, it only gets set once during the initial bootstrap.